src/tasks/MobileBuildTasks/Android/Ndk/Ndk.cs

```diff
@@ -0,0 +1,127 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+
+namespace Microsoft.Android.Build.Ndk
+{
+    public sealed class Ndk
+    {
+        private static string? ndkPath = "";
+        private static NdkVersion? ndkVersion;
+
+        public static string NdkPath
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(ndkPath))
+                {
+                    ndkPath = GetNdkPath(GetProbingPaths());
+                }
+
+                return ndkPath!;
+            }
+        }
+
+        public static NdkVersion NdkVersion
+        {
+            get
+            {
+#pragma warning disable IDE0074 // Use compound assignment
+                if (ndkVersion == null)
+                {
+                    ndkVersion = ReadVersion();
+                }
+#pragma warning restore IDE0074
+
+                return ndkVersion;
+            }
+        }
+
+        private static string? GetNdkPath(IEnumerable<string?> probingPaths)
+        {
+            string? ret = "";
+
+            foreach(string? path in probingPaths)
+            {
+                if (!string.IsNullOrEmpty(path) && Directory.Exists(path))
+                {
+                    ret = path;
+                    break;
+                }
+            }
+
+            return ret;
+        }
+
+        private static List<string?> GetProbingPaths()
```

Comment:
I'd do the null checks here so you can return `List<string>` and make your life easier in other methods